kit/src/components/embeds/link_embed/get_link_data.rs

```diff
@@ -0,0 +1,110 @@
+use std::error::Error;
+
+use reqwest::Url;
+use scraper::{Html, Selector};
+
+pub async fn get_image_data(document: Html, meta_selector: Selector) -> Option<String> {
```

Comment:
might want to break out  of the loop when the first image is detected. otherwise the last image will be returned.